hypershift-operator/metrics.go

```diff
@@ -23,6 +31,15 @@ const (
 	tickerTime = 30
 )
 
+var (
+	image                  string
```

Comment:
Having this global declaration and then in parallel calling `newMetrics` which reads them and `getOperatorImage` which writes them result in a race which is what you're seeing in the unit tests
```
WARNING: DATA RACE
Write at 0x0000051fd330 by goroutine 58:
  github.com/openshift/hypershift/hypershift-operator.getOperatorImage()
      /go/src/github.com/openshift/hypershift/hypershift-operator/metrics.go:576 +0x76d
  github.com/openshift/hypershift/hypershift-operator.TestOperatorInfo.func1()
      /go/src/github.com/openshift/hypershift/hypershift-operator/metrics_test.go:231 +0x31c
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1493 +0x47

Previous read at 0x0000051fd330 by goroutine 76:
  github.com/openshift/hypershift/hypershift-operator.newMetrics()
      /go/src/github.com/openshift/hypershift/hypershift-operator/metrics.go:142 +0x1310
  github.com/openshift/hypershift/hypershift-operator.TestMetrics.func4()
      /go/src/github.com/openshift/hypershift/hypershift-operator/metrics_test.go:119 +0x59d
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1493 +0x47

Goroutine 58 (running) created at:
  testing.(*T).Run()
      /usr/lib/golang/src/testing/testing.go:1493 +0x75d
  github.com/openshift/hypershift/hypershift-operator.TestOperatorInfo()
      /go/src/github.com/openshift/hypershift/hypershift-operator/metrics_test.go:226 +0xadc
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1493 +0x47

Goroutine 76 (running) created at:
  testing.(*T).Run()
      /usr/lib/golang/src/testing/testing.go:1493 +0x75d
  github.com/openshift/hypershift/hypershift-operator.TestMetrics()
      /go/src/github.com/openshift/hypershift/hypershift-operator/metrics_test.go:104 +0x1bde
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /usr/lib/golang/src/testing/testing.go:1493 +0x47
```

Please remove the global variables and let your function return the values. Then you can inove it locally where needed to set them.